src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueOfElement.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            Debug.Assert(value.ValueKind is JsonValueKind.False or JsonValueKind.True or JsonValueKind.Number or JsonValueKind.String);
+        }
+
+        private protected override JsonValueKind GetValueKindCore() => Value.ValueKind;
+        internal override JsonNode DeepCloneCore() => new JsonValueOfElement(Value.Clone(), Options);
+
+        internal override bool DeepEqualsCore(JsonNode? otherNode)
+        {
+            if (otherNode is JsonValueOfElement { Value: JsonElement otherElement })
+            {
+                JsonElement thisElement = Value;
+                if (thisElement.ValueKind != otherElement.ValueKind)
+                {
+                    return false;
+                }
+
+                switch (thisElement.ValueKind)
+                {
+                    case JsonValueKind.Null:
+                    case JsonValueKind.True:
+                    case JsonValueKind.False:
+                        return true;
+
+                    case JsonValueKind.String:
+                        if (otherElement.ValueIsEscaped)
+                        {
+                            if (thisElement.ValueIsEscaped)
+                            {
+                                // Both values contain escaping, force an allocation to unescape the RHS.
+                                return thisElement.ValueEquals(otherElement.GetString());
+                            }
+
+                            // Swap values so that unescaping is handled by the LHS.
+                            (thisElement, otherElement) = (otherElement, thisElement);
+                        }
+
+                        return thisElement.ValueEquals(otherElement.ValueSpan);
+
+                    case JsonValueKind.Number:
+                        return thisElement.GetRawValue().Span.SequenceEqual(otherElement.GetRawValue().Span);
+                    default:
+                        Debug.Fail("Object and Array JsonElements cannot be contained in JsonValue.");
+                        break;
+                }
+            }
+
+            return base.DeepEqualsCore(otherNode);
+        }
+
+        public override TypeToConvert GetValue<TypeToConvert>()
+        {
+            if (!TryGetValue(out TypeToConvert? value))
+            {
+                ThrowHelper.ThrowInvalidOperationException_NodeUnableToConvertElement(Value.ValueKind, typeof(TypeToConvert));
+            }
+
+            return value;
+        }
+
+        public override bool TryGetValue<TypeToConvert>([NotNullWhen(true)] out TypeToConvert value)
+        {
+            bool success;
+
+            if (Value is TypeToConvert element)
+            {
+                value = element;
+                return true;
+            }
+
+            switch (Value.ValueKind)
+            {
+                case JsonValueKind.Number:
+                    if (typeof(TypeToConvert) == typeof(int) || typeof(TypeToConvert) == typeof(int?))
```

Comment:
```suggestion
            Type t = Nullable.GetUnderlyingType(typeof(TypeToConvert)) ?? typeof(TypeToConvert);
            switch (Value.ValueKind)
            {
                case JsonValueKind.Number:
                    if (t == typeof(int))
```
Avoid so many ORs?

Reply:
This is simply copying existing code from a different source file. There are benefits in this verbosity since it lets the JIT perform branch elimination, see https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzmAPgAQEwEYCwAUJgAwAEAygBYQBOADgDITAB0ASgK4wJRgCmAbiJEA2gCkoCAOL8Y/WlADGACgQBPevwD2AMxWwEASiMBdIpgDMZHGQDCAHgAqAPjJEA3kTI+y9RQBuEAj8ZE5kAPpBADacoQC8ZAAm/LoQnNEIAITChL7ueb4SUrLyiqoaWnoGPCbmhT5WZACqZACyDs0uKkYF+V4N+WRQumRqmjr6zb3xiZWTNcZk6Ohk89XTZLNrE9WGAPwm3kM+AycnmADsY9Mq2sAAVvxKxlEQsfw5xycAviKD+SuyVS6UyX0Gf0IPyAA===